aws/internal/service/cognitoidentityprovider/waiter/status.go

```diff
@@ -0,0 +1,28 @@
+package waiter
+
+import (
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/cognitoidentityprovider"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource"
+)
+
+// UserPoolDomainStatus fetches the Operation and its Status
+func UserPoolDomainStatus(conn *cognitoidentityprovider.CognitoIdentityProvider, domain string) resource.StateRefreshFunc {
+	return func() (interface{}, string, error) {
+		input := &cognitoidentityprovider.DescribeUserPoolDomainInput{
+			Domain: aws.String(domain),
+		}
+
+		output, err := conn.DescribeUserPoolDomain(input)
+
+		if err != nil {
+			return nil, "", err
```

Comment:
I suggest a pattern of

```go
const (
	userPoolDomainStatusNotFound = "NotFound"
	userPoolDomainStatusUnknown  = "Unknown"
)

...

		if err != nil {
			return nil, userPoolDomainStatusUnknown, err
		}

		if output == nil {
			return nil, userPoolDomainStatusNotFound, nil
		}
```

Although these statuses are never explicitly tested I think having a value other than `""` helps with logging etc.
Having the constant names lowercase ensures they're not used outside this package.